### ISSUE TITLE
Support "nullable" (OpenApi 3.0 only)

### ISSUE THREAD

**Issue by ahx** [ahx/openapi_first#32]
Currently OpenapiFirst does not support "nullable" of the OpenApi flavoured JSON Schema.

~~See https://github.com/davishmcclurg/json_schemer/issues/22~~

**Comment by ahx** [comment]
json_schemer is currently not a good place to impement something like that. 
Here's an idea for a new rubygem:

```ruby
require 'openapi-json-schema-converter'
OpenapiJsonSchemaConverter.new(openapi_json_schema).call(data) # => standard_json_schema
```

**Comment by tristandruyen** [comment]
@ahx 
FYI this is a workaround we found:
```yaml
some_thing:
  oneOf:
    - type: 'null'   # Note the quotes around 'null'
    - type: string
```

which could be improved once OpenAPI 3.1 (#46) is supported to:

```yaml
some_thing:
  type: ['null',string]
```

so does this even need to stay open?

**Comment by ahx** [comment]
`type: ['null',string]` already works, even though it is not officially supported by OAS 3.0.

I think supporting `nullable` would be useful, because we could use standard OAS linters that would fail when using `type: ['null',string]`. OAS 3.1 solves all this, but unfortunately it has not reached RC status, yet.

**Comment by ahx** [comment]
This should be obsolete with OAS 3.1, since [nullable will be removed from OpenAPI](https://github.com/OAI/OpenAPI-Specification/pull/2246) and type arrays will be valid `type: ['null', string ]` 👍. 
I think we can close this.

**Comment by ahx** [comment]
Reopening, because I want to support OpenAPI 3.0.x

**Comment by ahx** [comment]
Reopening, because current implementation does not work with `allOf`, `oneOf` :|

**Comment by ahx** [comment]
@JF-Lalonde nullable has been [removed](https://github.com/OAI/OpenAPI-Specification/pull/2246) from OAS since 3.1. Is updating your spec to 3.1 an option for you/your team?

**Comment by JF-Lalonde** [comment]
@ahx Good to know! We've actually adopted the `type: ['null',string]` already even though it's not officially supported. But we'll probably be upgrading to 3.1 soon.

**Comment by ahx** [comment]
FYI this will be solved in 1.0, because json_schemer supports it with it's OpenAPI 3.0 / 3.1 metaschema
